### DISCUSSION TITLE
Some users gets no results when using wildcard or prefix

### DISCUSSION THREAD

**Discussion by ptrdvrs** [microsoft-search/pnp-modern-search#3865] [Q&A] [answered]
Using V3 for long time with no issues, but last week some users don't get any results. This issue is related to the login, because it moves with the user when login another laptop (inprivate session).
Installed V4 and the results are the same, some users gets no results.
I not sure if it's the search tool or something with the accounts, but hope someone can help. When using the default search bar on the top, the user gets results (in default SharePoint results screen).
Browser: edge
All tests are done in V4.12.2

No results for some users, other users get the good results:
{searchTerms}* AND contentclass="STS_List_DocumentLibrary" AND Path:{Site} AND -Title:"vervallen*" AND SiteTitle:"Opdracht_*"
Results for all users:
{searchTerms} AND contentclass="STS_List_DocumentLibrary" AND Path:{Site} AND -Title:"vervallen*" AND SiteTitle:"Opdracht_*"
THIS one is 'solved' by removing the wildcard.

But I have also a search with a prefix (Opdracht_).
No results for some users, other users get the good results:
Opdracht_{searchTerms}* AND contentclass:"STS_Web" AND Path:{Site} 
Evertything I tried, does not work, like using {} extra or "" or without wildcard.

Hope someone can help.

**Comment by kasperbolarsen** [comment]
I would recommend that you take a look at the query send to the backend, you can see how to inspect it here : https://youtu.be/zZN_zMAosaY?si=X5_X7Aojlwc0r3f7

**Reply by ptrdvrs** [reply]
Thank you. .The results in the devtools are the same by the person where the query is working as where it's not working. Nor errors like in the video.

**Comment by wobba** [comment]
I recommend using `({searchTerms}*)` over `{searchTerms}*` to ensure terms are AND'ed with everything happening after. Also try to change `=` to `:` for contentclass. `:` is actually a tiny bit more performant as well so use it if you have unique values. The `_` in SiteTitle is likely tokenized away and can cause query issues.

And write `SiteTitle:"Opdracht_*"` as `SiteTitle:(Opdracht_*)`, assuming the underscore do work.

**Reply by ptrdvrs** [reply]
Thanks for the tip and making the query better!!
Unfortunately the same results (some has results some none).
The issue is on multiple site collections. I will do a re-index of one site collection. Maybe there is something with the index in combination with some accounts. The users have the correct rights (search in de normal bar works, and the work in the libraries).

**Comment by PieterVandendriessche** [comment, marked as answer]
@ptrdvrs I actually was facing a similar issue. Since you are using V4, can you take a look at the screenshot below and try the last option in your query template? (as described in documentation: https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/)
![image](https://github.com/user-attachments/assets/0d02d031-a1ea-47b4-a881-fb3b4bcb1025)

**Reply by PieterVandendriessche** [reply]
Small remark, this is if you are using searchbox ;)

**Reply by ptrdvrs** [reply]
Yes!!!!!! That was the solution. I've used the {inputQueryText}. Thank you very much. Sorry for the delayed reply, I was on vacation.
